crates/services/p2p/src/peer_report.rs

```diff
@@ -142,17 +193,30 @@ impl NetworkBehaviour for Behaviour {
         cx: &mut Context<'_>,
     ) -> Poll<ToSwarm<Self::ToSwarm, THandlerInEvent<Self>>> {
         if let Some(event) = self.pending_events.pop_front() {
-            return Poll::Ready(ToSwarm::GenerateEvent(event))
+            return Poll::Ready(event)
         }
 
-        if self.decay_interval.poll_tick(cx).is_ready() {
-            return Poll::Ready(ToSwarm::GenerateEvent(PeerReportEvent::PerformDecay))
+        if let Some((instant, peer_id)) = self.reserved_nodes_to_connect.front() {
+            if instant.elapsed() > Duration::from_secs(HEALTH_CHECK_INTERVAL_IN_SECONDS) {
+                let peer_id = *peer_id;
+                self.reserved_nodes_to_connect.pop_front();
+                let multiaddrs = self
```

Comment:
The actual fix for the main reconnection problem that I faced with the dev cluster. The initial DNS address was replaced with a real IP, but when the sentry crashed, the IP was another one. Using initial multiaddrs here allows you to reconnect and get a new IP again.

Reply:
We should probably have a comment somewhere here explaining this.